02-workflow-plan.Rmd

```diff
@@ -2,9 +2,34 @@
 ottrpal::set_knitr_image_path()
 ```
 
-# Defining a workflow plan: Introduce somatic mutation calling workflow
+# Defining a workflow plan
 
+Our WDL guide will center around building a workflow from scratch. As we build out this workflow step-by-step, you will see what strategies and resources are used to develop a workflow that is iterative, reproducible, and efficient in terms of time and resource used. The goal is to use this workflow to illustrate common lessons in writing WDL workflows.
 
-## The workflow
+## Somatic mutation calling workflow
 
-## The test data
\ No newline at end of file
+The workflow that we will build is a somatic mutation calling workflow on paired cancer-normal samples. Our workflow is inspired from a somatic mutation calling [workflow](https://github.com/FredHutch/tg-wdl-LILAC-workflow/blob/master/GATK4.1.4.0/GATK4.1.4.0_manta.wdl) by the [Berger Lab](https://research.fredhutch.org/berger/en.html) at the Fred Hutch Cancer Center, and a somatic mutation calling [workflow](https://gatk.broadinstitute.org/hc/en-us/articles/360035894731-Somatic-short-variant-discovery-SNVs-Indels-) by the GATK team at the Broad Institute. The workflow takes in pairs of cancer and normal samples from sequencing in the FASTQ format and aligns the sequencing reads of each sample to the hg19 human reference genome, marks PCR duplicates, and recalibrates base quality. Which a pair of cancer and normal samples have reached this step, the workflow performs somatic mutation calling on them.
+
+The workflow schematic:
+
+workflow-schematic-goes-here
+
+## Workflow testing strategy
+
+As we build out our workflow, how do we know it is running correctly besides getting a message such as "Workflow finished with status 'Succeeded'" or an exit code 0? In [software development](https://www.atlassian.com/continuous-delivery/software-testing), it is essential to test your code to see whether it generates the expected output given a specified input. This principle applies into bioinformatics workflow development also:
+
+1.  Unit Testing: We need to incorporate tests to ensure that each task we develop is correct.
+
+2.  End-to-end testing: When we connect all the tasks together to form a workflow, we test that the workflow running end-to-end is correct.
+
+Here are some guidelines for any form of testing:
+
+-   The data you use for testing is representative of "real" data.
+
+-   You have an expectation of what the resulting output is *before* you run your workflow on it. It can be as specific as a MD5 checksum, or vague such as a certain file format.
+
+-   The process is quick to run, at the minutes level. This often means using a small subset of actual data.
```

Comment:
I would change the first sentence to "The process is quick to run, ideally in the range of just a few minutes." 95% of the time you should aim for a quick run, but there are some rare cases where what you are testing truly cannot be properly tested end-to-end properly in a few minutes. I consider "tests run within a few minutes" to be very useful, but I don't want users to think that's as important as the data being representative.

(Example: One workflow I worked on took in [dbGaP](https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/about.cgi) data using a special "drs" protocol. The testing needed to be able to check that the drs protocol was working, and for a while there were no downsampled samples I could test the drs integration with.)

Reply:
Sounds good!